source/game/kart/KartMove.cc

```diff
@@ -129,67 +170,266 @@ void KartMove::calc() {
         state()->setBoost(false);
     }
 
+    if (state()->isMtBoost()) {
+        if (--m_mtBoostTimer < 1) {
+            state()->setMtBoost(false);
+            m_mtBoostTimer = 0;
+        }
+    }
+
     calcVehicleSpeed();
     calcAcceleration();
     calcRotation();
 }
 
 void KartMove::calcTop() {
-    m_up = state()->top();
+    f32 stabilizationFactor = 0.1f;
+    m_hasLandingDir = false;
 
-    f32 topDotZ = 0.8f - 6.0f * (EGG::Mathf::abs(state()->top().dot(componentZAxis())));
-    f32 scalar = std::min(0.8f, std::max(0.3f, topDotZ));
+    // TODO: Add if-statement checking for GROUND_START bitfield
 
-    if (state()->isTouchingGround()) {
-        m_smoothedUp += (state()->top() - m_smoothedUp) * scalar;
-        m_smoothedUp.normalise();
+    if (!state()->isGroundStart()) {
```

Comment:
This should be `if (state()->isGroundStart())` due to the generated assembly instructions:
```asm
cmpwi r3,0x0
beq 0x78 ; beq generates because != 0 is the if block and == 0 is the else block
         ; This is at least the case in CW, as far as I'm aware
```